src/components/Input/Input.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { IconDefinition } from "@fortawesome/fontawesome-svg-core";
+import { cn } from "../../lib/utils";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export interface InputProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {
+  startIcon?: IconDefinition;
+  endIcon?: IconDefinition;
+}
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ className, type, startIcon, endIcon, ...props }, ref) => {
+    const StartIcon = startIcon;
+    const EndIcon = endIcon;
+    return (
+      <div className="w-full relative">
+        {StartIcon ? (
+          <>
+            <div className="absolute left-2.5 top-1/2 transform -translate-y-1/2">
+              <FontAwesomeIcon icon={startIcon} className="text-gray-400" />
+            </div>
+
+            <input
+              type={type}
+              className={cn(
```

Comment:
Für eine bessere Lesbarkeit würde ich die Styles hier in gruppieren und mehrere Strings an die cn-Funktion übergeben. Ähnlich wie wir es beim Switch machen (`src/components/Switch/Switch.tsx`).

Reply:
Habe versucht das irgendwie zu gliedern, welche Klassen zusammenpassen :)